src/unix/tty.c

```diff
@@ -34,6 +34,32 @@
 #define IMAXBEL 0
 #endif
 
+#if defined(__PASE__)
+/* On IBM i PASE, for better compatibility with running interactive programs in
+ * a 5250 environment, isatty() will return true for the stdin/stdout/stderr
+ * streams created by QSH/QP2TERM.
+ *
+ * For more, see docs on PASE_STDIO_ISATTY in
+ * https://www.ibm.com/support/knowledgecenter/ssw_ibm_i_74/apis/pase_environ.htm
+ *
+ * This behavior causes problems for Node as it expects that if isatty() returns
+ * true that TTY ioctls will be supported by that fd (which is not an
+ * unreasonable expectation) and when they don't it crashes with assertion
+ * errors.
+ *
+ * Here, we create our own version of isatty() that uses ioctl() to identify
+ * whether the fd is *really* a TTY or not.
+ */
+static int isatty(int file) {
+  int result = ioctl(file, TXISATTY + 0x81, NULL);
+
+  if (result != 0 && errno != EBADF)
+      errno = ENOTTY;
```

Comment:
```suggestion
  int result;

  result = ioctl(file, TXISATTY + 0x81, NULL);
  if (result != 0 && errno != EBADF)
    errno = ENOTTY;
```
(and most libuv calls it `rc` rather than `result` but that's a minor thing.)

Reply:
Gotcha. I was trying to figure out what to call it and saw [this](https://github.com/libuv/libuv/blob/v1.x/src/unix/tty.c#L42) and followed suit. In hindsight, rc seems obvious. Pushed up the suggested changes.